File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        int result = 0;

        while (x != 0) {
            int digit = x % 10;
            x /= 10;

            if (result > 214748364 ||
                (result == 214748364 && digit > 7))
                return 0;

            if (result < -214748364 ||
                (result == -214748364 && digit < -8))
                return 0;

            result = result * 10 + digit;
        }

        return result;
    }
}

File: 7-reverse-integer/reverse-integer.py
class Solution(object):
    def reverse(self, x):
        sign = -1 if x < 0 else 1
        x = abs(x)

        result = 0

        while x > 0:
            digit = x % 10
            result = result * 10 + digit
            x //= 10

        result *= sign

        if result < -2147483648 or result > 2147483647:
            return 0

        return result